sdk/java/pulumi/src/main/java/com/pulumi/test/internal/PulumiTestInternal.java

```diff
@@ -271,8 +272,8 @@ public PulumiTestInternal.Builder standardLogger(Logger standardLogger) {
          * @return this {@link PulumiTestInternal.Builder}
          */
         public PulumiTestInternal.Builder config(Map<String, String> allConfig, Set<String> configSecretKeys) {
-            return internalConfig(new DeploymentImpl.Config(
-                    ImmutableMap.copyOf(allConfig), ImmutableSet.copyOf(configSecretKeys)
+            return internalConfig(name -> new ConfigInternal(
```

Comment:
I needed to introduce the `configFactory` because, the project name is available only after it is initialized in `build()` - it is an internal API change